Userland/Libraries/LibWasm/AbstractMachine/Operators.h

```diff
@@ -124,6 +139,38 @@ struct BitRotateRight {
     static StringView name() { return "rotate_right"sv; }
 };
 
+template<size_t VectorSize, template<typename> typename SetSign = MakeSigned>
+struct VectorAllTrue {
+    auto operator()(u128 c) const
+    {
+        using ElementType = NativeIntegralType<128 / VectorSize>;
+
+        auto vector = bit_cast<Native128ByteVectorOf<ElementType, SetSign>>(c);
+
+        for (size_t i = 0; i < VectorSize; ++i)
+            if (vector[i] == 0)
+                return 0;
+
+        return 1;
+    }
```

Comment:
You don't actually need to write this out, it's much easier (and _much_ faster with more elements) to actually use SIMD:
```suggestion
    auto operator()(u128 c) const
    {
        using ElementType = NativeIntegralType<128 / VectorSize>;

        auto any_false = bit_cast<Native128ByteVectorOf<ElementType, SetSign>>(c) == 0;
        return bit_cast<u128>(any_false) == 0;
    }
```

Here's a comparison with the best impl I could think of, this, and your impl:
- [i8](https://quick-bench.com/q/2KuynRSQmcMoAen6aLOfdsZcEQA)
- [i16](https://quick-bench.com/q/76vKVotcBUv-BOfC2JDUcBF2vfw)
- [i64](https://quick-bench.com/q/WnvVQ9XHjJmh6Rpd-PZhPJjub6M)